bots/elasticsearch/tasks.py

```diff
@@ -132,3 +138,79 @@ def serialize(self, source):
             'name': safe_substr(source.long_title),
             'short_name': safe_substr(source.name)
         }
+
+
+def check_counts_in_range(self, min_date, max_date):
+    partial_database_count = CreativeWork.objects.exclude(title='').exclude(is_deleted=True).filter(
+        date_created__range=[min_date, max_date]
+    ).count()
+    partial_es_count = self.es_client.count(
+        index=(self.es_index or settings.ELASTICSEARCH_INDEX),
+        doc_type='creativeworks',
+        body={
+            'query': {
+                'range': {
+                    'date_modified': {
+                        'gte': '{}'.format(min_date.isoformat()),
+                        'lte': '{}'.format(max_date.isoformat())
+                    }
+                }
+            }
+        }
+    )['count']
+    return (partial_database_count == partial_es_count, partial_database_count, partial_es_count)
+
+
+def get_date_range_parts(min_date, max_date):
+    middle_date = min_date.average(max_date)
+    return {
+        'first_half': {'min_date': min_date, 'max_date': middle_date},
+        'second_half': {'min_date': middle_date, 'max_date': max_date}
+    }
+
+
+@celery.task
+def pseudo_bisection_method(self, min_date, max_date):
```

Comment:
>  I figured the task doesn't need to be logged since there could be a lot of tasks and each one doesn't really mean anything.

Agreed

> self still needs to be passed along to preserve the elastic client but it seems weird to me.

`self` won't be JSON serializable.

You can reconstruct the Elasticsearch client and accept all the same options (`ES_URL`, `ES_INDEX`, etc)
Or build this logic into the main task.

I don't really have a preference, follow your ❤️